Test/Pester5.Tests.ps1

```diff
@@ -0,0 +1,563 @@
+                    'VerbosePreference',
+                    'WarningPreference'
+                ) | Sort-Object
+                $ArgumentList[4].Keys | Sort-Object | Should -Be $prefNames
+                return $true
+            }
+            finally
+            {
+                Pop-Location
+            }
+        }
+    }
+}
+
+Describe 'Pester5.when passing custom arguments' {
+    AfterEach { Reset }
+    It 'should pass the arguments' {
+        Init
+        GivenTestFile 'PassingTests.ps1' @'
+Describe 'PassingTests'{
+    It 'should pass' {
+        $true | Should -BeTrue
+    }
+}
+Describe 'FailingTests'{
+    It 'should fail' {
+        $false | Should -BeTrue
+    }
+}
+'@
+        WhenPesterTaskIsInvoked -AsJob -WithArgument @{
+            # Make sure the Pester5 task's default values for these get overwritten
+            'Output' = '.output\pester.xml';
+            'OutputFormat' = 'JUnitXml';
+            # Make sure these do *not* get overwritten.
+            'PassThru' = $false;
+            # Make sure this gets passed.
+            'TestName' = 'PassingTests';
+        }
+        ThenPesterShouldHaveRun -FailureCount 0 -PassingCount 1 -AsJUnitXml -ResultFileName 'pester.xml'
+    }
+}
+
+Describe 'Pester5.when passing named arguments to script' {
+    AfterEach { Reset }
+    It 'should pass arguments' {
+        Init
+        $oneValue = [Guid]::NewGuid()
+        $twoValue = [Guid]::NewGuid()
+        $threeValue = [Guid]::NewGuid()
+        $fourValue = [Guid]::NewGuid()
+        $testContent = @"
+param(
+    [Parameter(Mandatory,Position=0)]
+    [String]`$One,
+
+    [Parameter(Mandatory,Position=1)]
+    [String]`$Two,
+
+    [Parameter(Mandatory)]
+    [String]`$Three,
+
+    [Parameter(Mandatory)]
+    [String]`$Four
+)
+
+Describe 'ArgTest' {
+    It 'should pass them' {
+        `$One | Should -Be '$($oneValue)'
+        `$Two | Should -Be '$($twoValue)'
+        `$Three | Should -Be '$($threeValue)'
+        `$Four | Should -Be '$($fourValue)'
+    }
+}
+"@
+        GivenTestFile 'ArgTest.ps1' $testContent
+        GivenTestFile 'Arg2Test.ps1' $testContent
+        WhenPesterTaskIsInvoked -AsJob -WithArgument @{
+            'Script' = @{
+                'Path' = 'Arg*.ps1';
+                'Data' = @{
+                    One = $oneValue;
+                    Two = $twoValue;
+                    Three = $threeValue;
+                    Four = $fourValue;
+                };
+            }
+        }
+        ThenPesterShouldHaveRun -PassingCount 2 -FailureCount 0
+    }
+}
+
+Describe 'Pester5.when passing hashtable to script property with multiple paths' {
+    AfterEach { Reset }
+    It 'should pass arguments' {
+        Init
+        WhenPesterTaskIsInvoked -AsJob -WithArgument @{
+            'Script' = @{
+                'Path' = ('Path1.ps1','Path2.ps1');
+            };
+        } -ErrorAction SilentlyContinue
+        ThenPesterShouldHaveRun -PassingCount 0 -FailureCount 0
+    }
+}
```

Comment:
Make sure you have tests cases that test:

1. An array list gets converted correctly.
2. A string boolean gets converted correctly.
3. Passing containers works for: paths and script blocks.